pvlib/ivtools.py

```diff
@@ -262,6 +262,142 @@ def fit_sde_sandia(voltage, current, v_oc=None, i_sc=None, v_mp_i_mp=None,
+    irrad_ref: float, default 1000
+        Reference irradiance condition [W/m2]
+
+    Returns
+    -------
+    Tuple of the following elements:
+
+        * Dictionary with the following elements:
+            I_L_ref: float
+                Light-generated current at reference conditions [A]
+            I_o_ref: float
+                Diode saturation current at reference conditions [A]
+            R_s: float
+                Series resistance [ohms]
+            R_sh_ref: float
+                Shunt resistance at reference conditions [ohms].
+            a_ref: float
+                Modified ideality factor at reference conditions.
+                The product of the usual diode ideality factor (n, unitless),
+                number of cells in series (Ns), and cell thermal voltage at
+                specified effective irradiance and cell temperature.
+            alpha_sc: float
+                The short-circuit current (i_sc) temperature coefficient of the
+                module [A/K].
+            EgRef: float
+                Energy of bandgap of semi-conductor used [eV]
+            dEgdT: float
+                Variation of bandgap according to temperature [eV/K]
+            irrad_ref: float
+                Reference irradiance condition [W/m2]
+            temp_ref: float
+                Reference temperature condition [C]
+        * scipy.optimize.OptimizeResult
+            Optimization result of scipy.optimize.root().
+            See scipy.optimize.OptimizeResult for more details.
+
+    References
+    ----------
+    [1] W. De Soto et al., "Improvement and validation of a model for
+    photovoltaic array performance", Solar Energy, vol 80, pp. 78-88,
+    2006.
+
+    [2] John A Dufﬁe, William A Beckman, "Solar Engineering of Thermal
+    Processes", Wiley, 2013
+    """
+
+    try:
+        from scipy.optimize import root
+        from scipy import constants
+    except ImportError:
+        raise ImportError("The fit_sdm_desoto function requires scipy.")
+
+    # Constants
+    k = constants.value('Boltzmann constant in eV/K')
+    Tref = temp_ref + 273.15  # [K]
+
+    # initial guesses of variables for computing convergence:
+    # Values are taken from [2], p753
+    Rsh_i = 100.0
+    a_i = 1.5*k*Tref*cells_in_series
+    IL_i = i_sc
+    Io_i = i_sc * np.exp(-v_oc/a_i)
+    Rs_i = (a_i*np.log1p((IL_i-i_mp)/Io_i) - v_mp)/i_mp
+    # params_i : initial values vector
+    params_i = np.array([IL_i, Io_i, a_i, Rsh_i, Rs_i])
+
+    # specs of module
+    specs = ((i_sc, v_oc, i_mp, v_mp, beta_voc, alpha_sc, EgRef, dEgdT,
+              Tref, k),)
+
+    # computing with system of equations described in [1]
+    optimize_result = root(_system_of_equations, x0=params_i, args=specs)
+
+    if optimize_result.success:
+        sdm_params = optimize_result.x
+    else:
+        raise RuntimeError('Parameter estimation failed:\n'
+                           + optimize_result.message)
```

Comment:
I agree with stickler in this case - better to move the plus sign to the end of the line above

Reply:
Actually the plus sign was above in the previous commit I made, but stickler found a lint error as well. I choose to keep it below according to this:
https://www.python.org/dev/peps/pep-0008/#should-a-line-break-before-or-after-a-binary-operator

Reply:
I humbly suggest:
```python
        raise RuntimeError(
            'Parameter estimation failed:\n' + optimize_result.message)
```
which keeps the string on a single line and which I think is a short enough line.